library/ssl_tls.c

```diff
@@ -7910,14 +7907,12 @@ static int ssl_session_load_tls12( mbedtls_ssl_session *session,
     /*
      * Basic mandatory fields
      */
-    if( 2 + 1 + 1 + 32 + 48 + 4 > (size_t)( end - p ) )
+    if( 2 + 1 + 32 + 48 + 4 > (size_t)( end - p ) )
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
 
     session->ciphersuite = ( p[0] << 8 ) | p[1];
     p += 2;
 
-    session->compression = *p++;
```

Comment:
This does not fulfill the requirement to load old sessions for compatibility's sake. The session format should be the same, but this field - ignored.